test/v1/connectors/ListBenefitsConnectorSpec.scala

```diff
@@ -94,21 +107,42 @@ class ListBenefitsConnectorSpec extends ConnectorSpec {
 
   "ListBenefitsConnector" when {
     "listBenefits" when {
+      "a benefitId query param is provided" must {
+        "return a 200 status for a success scenario" in new IfsTest with Test {
+          val request = ListBenefitsRequest(Nino(nino),taxYear="2019-20",Some(benefitId))
+          val outcome = Right(ResponseWrapper(correlationId, validResponse))
+
+          willGet(s"$baseUrl/income-tax/income/state-benefits/$nino/$taxYear") returns Future.successful(outcome)
+
+          val result = await(connector.listBenefits(request))
+
+          result shouldBe outcome
+        }
+      }
+
       "no benefitId query param is provided" must {
-        "return a 200 status for a success scenario" in new Test {
+        "return a 200 status for a success scenario" in new IfsTest with Test {
+          val request = ListBenefitsRequest(Nino(nino),taxYear="2019-20",None)
           val outcome = Right(ResponseWrapper(correlationId, validResponse))
-          stubHttp(outcome, Seq("benefitId" -> benefitId))
 
-          await(connector.listBenefits(ListBenefitsRequest(Nino(nino), taxYear, Some(benefitId)))) shouldBe outcome
+          willGet(s"$baseUrl/income-tax/income/state-benefits/$nino/$taxYear") returns Future.successful(outcome)
```

Comment:
What makes you prefer ```stubTYSHttp```? Particularly since higher up (line 91) it seems you suggest replacing ```stubTYSHttp``` with ```willGet```?

Reply:
I think the whole purpose of the stubTYSHttp is to avoid having to repeatedly use the willGet method and repeatedly use what will be the same URI. TBH, either are fine as long as it is consistently used throughout the test class, but no point having the stubTYSHttp if its not going to be used